monkestation/code/game/objects/items/anime.dm

```diff
@@ -11,37 +11,15 @@
 	var/food_dislikes
 	var/list/weeb_screams
 	var/list/weeb_laughs
+	var/hacked = FALSE
 
 /obj/item/anime/attack_self(mob/living/carbon/user)
 	if(ishuman(user))
-		var/mob/living/carbon/human/weeb = user
-		var/new_color = input(user, "Choose an Anime color:", "Anime Color (clicking 'cancel' will set Anime color to Hair color):", weeb.hair_color) as color|null
-		if(new_color) //If they DON'T pick a color, then it just defaults to their original hair color.
-			weeb.custom_color = sanitize_hexcolor(new_color)
-		else
-			weeb.custom_color = weeb.hair_color
-		if(ears)
-			ears.Insert(weeb)
-		if(tail)
-			tail.Insert(weeb)
-		if(weeb_screams)
-			weeb.alternative_screams += weeb_screams
-		if(weeb_laughs)
-			weeb.alternative_laughs += weeb_laughs
-		if(food_likes)
-			weeb.dna.species.liked_food = food_likes
-		if(food_dislikes)
-			weeb.dna.species.disliked_food = food_dislikes
-			weeb.dna.species.toxic_food = food_dislikes
+		anime_transformation(user)
 
-		var/turf/location = get_turf(weeb)
-		weeb.add_splatter_floor(location)
-		var/msg = "<span class=danger>You feel the power of God and Anime flow through you!</span>"
-		to_chat(weeb, msg)
-		playsound(location, 'sound/weapons/circsawhit.ogg', 50, 1)
-		weeb.update_body()
-		weeb.update_hair()
-		qdel(src)
+/obj/item/anime/afterattack(mob/living/carbon/user)
+	if(hacked == TRUE && ishuman(user))
+		anime_transformation(user)
```

Comment:
I think you're on the right track, turning it into a proc.
Though won't this cause the user and not the target to get changed?

Reply:
the first argument in afterattack is atom/target. I dont think naming it user makes it correspond to the user argument, so it's probably just weirdly named and should be mob/living/carbon/target or something similar I think

Reply:
yeah I thought that might happen but I tested and it works fine.